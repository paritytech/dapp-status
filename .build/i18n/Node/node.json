[
  {
    "id": "status.status.title.bestBlock",
    "defaultMessage": "best block"
  },
  {
    "id": "status.status.title.peers",
    "defaultMessage": "peers"
  },
  {
    "id": "status.status.title.hashRate",
    "defaultMessage": "hash rate"
  },
  {
    "id": "status.status.hashrate",
    "defaultMessage": "{hashrate} H/s"
  },
  {
    "id": "status.status.title.node",
    "defaultMessage": "Node"
  },
  {
    "id": "status.status.title.network",
    "defaultMessage": "network settings"
  },
  {
    "id": "status.status.input.chain",
    "defaultMessage": "chain"
  },
  {
    "id": "status.status.input.rpcEnabled",
    "defaultMessage": "rpc enabled"
  },
  {
    "id": "status.status.input.yes",
    "defaultMessage": "yes"
  },
  {
    "id": "status.status.input.no",
    "defaultMessage": "no"
  },
  {
    "id": "status.status.input.port",
    "defaultMessage": "network port"
  },
  {
    "id": "status.status.input.rpcInterface",
    "defaultMessage": "rpc interface"
  },
  {
    "id": "status.status.input.rpcPort",
    "defaultMessage": "rpc port"
  },
  {
    "id": "status.status.input.enode",
    "defaultMessage": "enode"
  }
]