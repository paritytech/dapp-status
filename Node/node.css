/* Copyright 2015-2017 Parity Technologies (UK) Ltd.
/* This file is part of Parity.
/*
/* Parity is free software: you can redistribute it and/or modify
/* it under the terms of the GNU General Public License as published by
/* the Free Software Foundation, either version 3 of the License, or
/* (at your option) any later version.
/*
/* Parity is distributed in the hope that it will be useful,
/* but WITHOUT ANY WARRANTY; without even the implied warranty of
/* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
/* GNU General Public License for more details.
/*
/* You should have received a copy of the GNU General Public License
/* along with Parity.  If not, see <http://www.gnu.org/licenses/>.
*/

.container {
  padding-bottom: 1em;
}

.row {
  margin: 0 -1em;

  &::after {
    display: table;
    clear: both;
    content: '';
  }
}

.blockInfo {
  color: #aaa;
  font-size: 1.5em;
  line-height: 1.5em;
}

.blockByline {
  color: #aaa;
  font-size: 0.75em;
}

.padBottom {
  padding-bottom: 1.25em !important;
}

.col,
.col3, .col4, .col4_5, .col6, .col12 {
  float: left;
  padding: 0 1em;
  box-sizing: border-box;
}

.col3 {
  width: 25%;
  width: -webkit-calc(100% / 12 * 3);
  width: -moz-calc(100% / 12 * 3);
  width: calc(100% / 12 * 3);
}

.col4 {
  width: 33.3%;
  width: -webkit-calc(100% / 12 * 4);
  width: -moz-calc(100% / 12 * 4);
  width: calc(100% / 12 * 4);
}

.col4_5 {
  width: 37.5%;
  width: -webkit-calc(100% / 12 * 4.5);
  width: -moz-calc(100% / 12 * 4.5);
  width: calc(100% / 12 * 4.5);
}

.col6 {
  width: 50%;
  width: -webkit-calc(100% / 12 * 6);
  width: -moz-calc(100% / 12 * 6);
  width: calc(100% / 12 * 6);
}

.col12 {
  width: 100%;
}
