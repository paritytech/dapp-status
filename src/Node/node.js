// Copyright 2015-2017 Parity Technologies (UK) Ltd.
// This file is part of Parity.

// Parity is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.

// Parity is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.

// You should have received a copy of the GNU General Public License
// along with Parity.  If not, see <http://www.gnu.org/licenses/>.

import bytes from 'bytes';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { observer } from 'mobx-react';

import { BlockNumber, BlockTimestamp, Container, ContainerTitle, Input, NetPeers } from '@parity/ui';

import MiningSettings from '../MiningSettings';
import StatusStore from './store';

import styles from './node.css';

@observer
export default class Node extends Component {
  static contextTypes = {
    api: PropTypes.object.isRequired
  };

  statusStore = new StatusStore(this.context.api);

  render () {
    const { hashrate } = this.statusStore;
    const hashrateValue = bytes(hashrate.toNumber()) || 0;

    return (
      <Container>
        <div className={ styles.container }>
          <div className={ styles.row }>
            <div className={ styles.col3 }>
              <div className={ `${styles.col12} ${styles.padBottom}` }>
                <ContainerTitle
                  title={
                    <FormattedMessage
                      id='status.status.title.bestBlock'
                      defaultMessage='best block'
                    />
                  }
                />
                <div className={ styles.blockInfo }>
                  #<BlockNumber />
                </div>
                <div className={ styles.blockByline }>
                  <BlockTimestamp />
                </div>
              </div>
              <div className={ `${styles.col12} ${styles.padBottom}` }>
                <ContainerTitle
                  title={
                    <FormattedMessage
                      id='status.status.title.peers'
                      defaultMessage='peers'
                    />
                  }
                />
                <div className={ styles.blockInfo }>
                  <NetPeers />
                </div>
              </div>
              <div className={ `${styles.col12} ${styles.padBottom}` }>
                <ContainerTitle
                  title={
                    <FormattedMessage
                      id='status.status.title.hashRate'
                      defaultMessage='hash rate'
                    />
                  }
                />
                <div className={ styles.blockInfo }>
                  <FormattedMessage
                    id='status.status.hashrate'
                    defaultMessage='{hashrate} H/s'
                    values={ {
                      hashrate: hashrateValue
                    } }
                  />
                </div>
              </div>
            </div>
            <div className={ styles.col4_5 }>
              { this.renderMiningSettings() }
            </div>
            <div className={ styles.col4_5 }>
              { this.renderSettings() }
            </div>
          </div>
        </div>
      </Container>
    );
  }

  renderMiningSettings () {
    const { coinbase, defaultExtraData, extraData, gasFloorTarget, minGasPrice } = this.statusStore;

    return (
      <MiningSettings
        coinbase={ coinbase }
        defaultExtraData={ defaultExtraData }
        extraData={ extraData }
        gasFloorTarget={ gasFloorTarget }
        minGasPrice={ minGasPrice }
        onUpdateSetting={ this.statusStore.handleUpdateSetting }
      />
    );
  }

  renderNodeName () {
    const { nodeName } = this.statusStore;

    return (
      <span>
        { nodeName || (
          <FormattedMessage
            id='status.status.title.node'
            defaultMessage='Node'
          />)
        }
      </span>
    );
  }

  renderSettings () {
    const { enode, rpcSettings, netPort = '' } = this.statusStore;

    if (!rpcSettings) {
      return null;
    }

    const rpcPort = rpcSettings.port || '';

    return (
      <div>
        <ContainerTitle
          title={
            <FormattedMessage
              id='status.status.title.network'
              defaultMessage='network settings'
            />
          }
        />
        <Input
          allowCopy
          readOnly
          label={
            <FormattedMessage
              id='status.status.input.chain'
              defaultMessage='chain'
            />
          }
          value={ this.statusStore.netChain }
        />
        <div className={ styles.row }>
          <div className={ styles.col6 }>
            <Input
              allowCopy
              readOnly
              label={
                <FormattedMessage
                  id='status.status.input.rpcEnabled'
                  defaultMessage='rpc enabled'
                />
              }
              value={
                rpcSettings.enabled
                  ? (
                    <FormattedMessage
                      id='status.status.input.yes'
                      defaultMessage='yes'
                    />
                  )
                  : (
                    <FormattedMessage
                      id='status.status.input.no'
                      defaultMessage='no'
                    />
                  )
              }
            />
          </div>
          <div className={ styles.col6 }>
            <Input
              allowCopy
              readOnly
              label={
                <FormattedMessage
                  id='status.status.input.port'
                  defaultMessage='network port'
                />
              }
              value={ netPort.toString() }
            />
          </div>
        </div>

        <div className={ styles.row }>
          <div className={ styles.col6 }>
            <Input
              allowCopy
              readOnly
              label={
                <FormattedMessage
                  id='status.status.input.rpcInterface'
                  defaultMessage='rpc interface'
                />
              }
              value={ rpcSettings.interface }
            />
          </div>
          <div className={ styles.col6 }>
            <Input
              allowCopy
              readOnly
              label={
                <FormattedMessage
                  id='status.status.input.rpcPort'
                  defaultMessage='rpc port'
                />
              }
              value={ rpcPort.toString() }
            />
          </div>
        </div>

        <div className={ styles.row }>
          <div className={ styles.col12 }>
            <Input
              allowCopy
              readOnly
              label={
                <FormattedMessage
                  id='status.status.input.enode'
                  defaultMessage='enode'
                />
              }
              value={ enode }
            />
          </div>
        </div>
      </div>
    );
  }
}
