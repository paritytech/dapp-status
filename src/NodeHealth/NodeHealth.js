// Copyright 2015-2017 Parity Technologies (UK) Ltd.
// This file is part of Parity.

// Parity is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.

// Parity is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.

// You should have received a copy of the GNU General Public License
// along with Parity.  If not, see <http://www.gnu.org/licenses/>.

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { inject, observer } from 'mobx-react';

import { STATUS_OK, STATUS_BAD } from '@parity/mobx/lib/node/NodeHealthStore';
import StatusIndicator from '@parity/ui/lib/StatusIndicator';
import Statistic from 'semantic-ui-react/dist/commonjs/views/Statistic';
import { FormattedMessage } from 'react-intl';

import Section from '../components/Section';
import styles from './NodeHealth.css';

export class NodeHealth extends Component {
  static propTypes = {
    nodeHealthStore: PropTypes.object.isRequired
  };

  render() {
    const { nodeHealthStore: { health, overall } } = this.props;

    return (
      <Section
        title={
          <FormattedMessage
            id="dapp.status.nodeHealthTitle"
            defaultMessage="Node Health"
          />
        }
      >
        <Statistic.Group widths={4}>
          <Statistic>
            <Statistic.Value>
              {health.sync && health.sync.status === STATUS_OK ? (
                <FormattedMessage id="dapp.status.sync" defaultMessage="sync" />
              ) : (
                <FormattedMessage
                  id="dapp.status.notSync"
                  defaultMessage="not sync"
                />
              )}
            </Statistic.Value>
            <Statistic.Label>
              {health.sync && health.sync.status === STATUS_OK ? (
                <FormattedMessage
                  id="dapp.status.chainSynchronized"
                  defaultMessage="Chain Synchronized"
                />
              ) : (
                <FormattedMessage
                  id="dapp.status.chainNotSynchronized"
                  defaultMessage="Chain Not Synchronized"
                />
              )}
            </Statistic.Label>
          </Statistic>
          <Statistic>
            <Statistic.Value>
              {health.peers
                ? `${health.peers.details[0]}/${health.peers.details[1]}`
                : '0/25'}
            </Statistic.Value>
            <Statistic.Label>
              <FormattedMessage
                id="dapp.status.connectedPeers"
                defaultMessage="Connected Peers"
              />
            </Statistic.Label>
          </Statistic>
          <Statistic>
            <Statistic.Value>
              {health.time ? (
                <span>
                  {health.time.details}{' '}
                  <span className={styles.timeUnity}>ms</span>
                </span>
              ) : (
                '-'
              )}
            </Statistic.Value>
            <Statistic.Label>
              {health.time && health.time.status === STATUS_OK ? (
                <FormattedMessage
                  id="dapp.status.timeSynchronized"
                  defaultMessage="Time Synchronized"
                />
              ) : (
                <FormattedMessage
                  id="dapp.status.timeNotSynchronized"
                  defaultMessage="Time Not Synchronized"
                />
              )}
            </Statistic.Label>
          </Statistic>
          <Statistic>
            <Statistic.Value>
              <StatusIndicator
                id="overall"
                status={overall ? overall.status : STATUS_BAD}
              />
            </Statistic.Value>
            <Statistic.Label>
              <FormattedMessage
                id="dapp.status.overallHealth"
                defaultMessage="Overall"
              />
            </Statistic.Label>
          </Statistic>
        </Statistic.Group>
      </Section>
    );
  }
}

export default inject('nodeHealthStore')(observer(NodeHealth));
