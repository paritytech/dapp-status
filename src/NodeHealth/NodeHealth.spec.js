// Copyright 2015-2017 Parity Technologies (UK) Ltd.
// This file is part of Parity.

// Parity is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.

// Parity is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.

// You should have received a copy of the GNU General Public License
// along with Parity.  If not, see <http://www.gnu.org/licenses/>.

import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { STATUS_OK } from '@parity/mobx/lib/node/NodeHealthStore';

import { NodeHealth } from './NodeHealth';

const props = {
  nodeHealthStore: {
    health: {
      peers: { status: STATUS_OK, details: [2, 3] },
      sync: { status: STATUS_OK },
      time: { status: STATUS_OK, details: 'Foo' }
    },
    overall: STATUS_OK
  }
};

test('should render correctly', () => {
  const component = shallow(<NodeHealth {...props} />);

  expect(shallowToJson(component)).toMatchSnapshot();
});

test('should render correctly not fully loaded', () => {
  const component = shallow(<NodeHealth nodeHealthStore={{ health: {} }} />);

  expect(shallowToJson(component)).toMatchSnapshot();
});
