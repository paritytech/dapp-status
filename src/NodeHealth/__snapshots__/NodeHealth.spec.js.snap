// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Section
  title={
    <FormattedMessage
      defaultMessage="Node Health"
      id="dapp.status.nodeHealthTitle"
      values={Object {}}
    />
  }
>
  <StatisticGroup
    widths={4}
  >
    <Statistic>
      <StatisticValue>
        <FormattedMessage
          defaultMessage="sync"
          id="dapp.status.sync"
          values={Object {}}
        />
      </StatisticValue>
      <StatisticLabel>
        <FormattedMessage
          defaultMessage="Chain Synchronized"
          id="dapp.status.chainSynchronized"
          values={Object {}}
        />
      </StatisticLabel>
    </Statistic>
    <Statistic>
      <StatisticValue>
        2/3
      </StatisticValue>
      <StatisticLabel>
        <FormattedMessage
          defaultMessage="Connected Peers"
          id="dapp.status.connectedPeers"
          values={Object {}}
        />
      </StatisticLabel>
    </Statistic>
    <Statistic>
      <StatisticValue>
        <span>
          Foo
           
          <span>
            ms
          </span>
        </span>
      </StatisticValue>
      <StatisticLabel>
        <FormattedMessage
          defaultMessage="Time Synchronized"
          id="dapp.status.timeSynchronized"
          values={Object {}}
        />
      </StatisticLabel>
    </Statistic>
    <Statistic>
      <StatisticValue>
        <StatusIndicator
          id="overall"
        />
      </StatisticValue>
      <StatisticLabel>
        <FormattedMessage
          defaultMessage="Overall"
          id="dapp.status.overallHealth"
          values={Object {}}
        />
      </StatisticLabel>
    </Statistic>
  </StatisticGroup>
</Section>
`;

exports[`should render correctly not fully loaded 1`] = `
<Section
  title={
    <FormattedMessage
      defaultMessage="Node Health"
      id="dapp.status.nodeHealthTitle"
      values={Object {}}
    />
  }
>
  <StatisticGroup
    widths={4}
  >
    <Statistic>
      <StatisticValue>
        <FormattedMessage
          defaultMessage="not sync"
          id="dapp.status.notSync"
          values={Object {}}
        />
      </StatisticValue>
      <StatisticLabel>
        <FormattedMessage
          defaultMessage="Chain Not Synchronized"
          id="dapp.status.chainNotSynchronized"
          values={Object {}}
        />
      </StatisticLabel>
    </Statistic>
    <Statistic>
      <StatisticValue>
        0/25
      </StatisticValue>
      <StatisticLabel>
        <FormattedMessage
          defaultMessage="Connected Peers"
          id="dapp.status.connectedPeers"
          values={Object {}}
        />
      </StatisticLabel>
    </Statistic>
    <Statistic>
      <StatisticValue>
        -
      </StatisticValue>
      <StatisticLabel>
        <FormattedMessage
          defaultMessage="Time Not Synchronized"
          id="dapp.status.timeNotSynchronized"
          values={Object {}}
        />
      </StatisticLabel>
    </Statistic>
    <Statistic>
      <StatisticValue>
        <StatusIndicator
          id="overall"
          status="bad"
        />
      </StatisticValue>
      <StatisticLabel>
        <FormattedMessage
          defaultMessage="Overall"
          id="dapp.status.overallHealth"
          values={Object {}}
        />
      </StatisticLabel>
    </Statistic>
  </StatisticGroup>
</Section>
`;
